src/components/encounter/encounter-form-manager.ts

```diff
@@ -106,6 +127,38 @@ export class EncounterFormManager {
       return savePatientIdentifier(patientIdentifier, patient.id);
     });
   }
+
+  static saveProgramEnrollments = (programPayload: any, sessionMode: string, patient: any) => {
+    const ac = new AbortController();
+    if (programPayload.program != 'undefined') {
+      let patientProgramEnrollment = patient.patientPrograms.find(
+        (enrollment) => enrollment.program.uuid === programPayload.program && enrollment.dateCompleted === null,
+      );
+
+      // return already enrolled if not edit mode
+      if (sessionMode == 'edit' && patientProgramEnrollment.uuid !== '') {
+        let previousPatientProgramEnrollment;
+        let currentPatientEnrollment; 
+
+        // invalidate previous and update new
+        if (!programPayload.dateEnrolled) {
+          programPayload.dateEnrolled = patientProgramEnrollment.dateEnrolled;
```

Comment:
```suggestion
          programPayload.dateEnrolled = patientProgramEnrollment?.dateEnrolled;
```
Failsafe incase `patientProgramEnrollment` does not resolve.